packages/adapters-library/src/adapters/solv/products/yield-market/solvYieldMarketAdapter.ts

```diff
@@ -266,159 +277,146 @@ export class SolvYieldMarketAdapter implements IProtocolAdapter {
-      await this.openFundRedemptionDelegateContract.tokenOfOwnerByIndex(
-        userAddress,
-        index,
-      )
-    const balance =
-      await this.openFundRedemptionDelegateContract['balanceOf(uint256)'](
-        tokenId,
-      )
+    sftType: 'share' | 'redemption' = 'share',
+  ): Promise<bigint[]> {
+    const contract =
+      sftType === 'share'
+        ? this.openFundShareDelegateContract
+        : this.openFundRedemptionDelegateContract
 
-    if (!balance) return
-
-    const decimals =
-      await this.openFundRedemptionDelegateContract.valueDecimals()
-    const slot = await this.openFundRedemptionDelegateContract.slotOf(tokenId)
-    const [poolId, currency] =
-      await this.openFundRedemptionConcreteContract.getRedeemInfo(slot)
-    const [latestSetNavTime] = await this.navOracleContract.poolNavInfos(poolId)
-    const [nav] = await this.navOracleContract.getSubscribeNav(
-      poolId,
-      latestSetNavTime,
-    )
+    const tokensCount = await contract['balanceOf(address)'](userAddress)
 
-    const shareSlot = this.slotReverseLookup(poolId)
-    const name = `${this.getPoolName(
-      shareSlot.toString(),
-    )} | Redemption pending`
+    if (!tokensCount) return []
 
-    const {
-      symbol: underlyingSymbol,
-      decimals: underlyingDecimals,
-      name: underlyingName,
-    } = await this.helpers.getTokenMetadata(currency)
+    const indexes = [...Array(Number.parseInt(tokensCount.toString())).keys()]
 
-    const position: ProtocolPosition = {
-      type: TokenType.Protocol,
-      balanceRaw: balance,
-      address: sftAddress,
-      tokenId: tokenId.toString(),
-      name,
-      symbol: name,
-      decimals: Number(decimals),
-      tokens: [
-        {
-          type: TokenType.Underlying,
-          address: currency,
-          priceRaw: nav,
-          balanceRaw: (balance * nav) / 10n ** decimals,
-          name: underlyingName,
-          symbol: underlyingSymbol,
-          decimals: Number(underlyingDecimals),
-        },
-      ],
-    }
+    const tokenIds = await Promise.all(
+      indexes.map((index) => contract.tokenOfOwnerByIndex(userAddress, index)),
+    )
 
-    return position
+    return tokenIds
   }
 
   /**
    * Computes a Pool ID from the SFT address and the slot.
    * Some Solv smart contracts methods accept the pool id instead of the slot.
    *
    * @example
-   * const openFundShareDelegateAddress = '0x22799daa45209338b7f938edf251bdfd1e6dcb32'
    * const slot = '5310353805259224968786693768403624884928279211848504288200646724372830798580'
-   * computePoolId(sftAddress, slot) // returns '0xe037ef7b5f74bf3c988d8ae8ab06ad34643749ba9d217092297241420d600fce'
+   * computePoolId(slot) // returns '0xe037ef7b5f74bf3c988d8ae8ab06ad34643749ba9d217092297241420d600fce'
    *
    * @see https://github.com/solv-finance/SolvBTC/blob/ef5be00ec22549ac5a323378c2a914166bf0dcc1/contracts/SftWrappedToken.sol#L198
    */
   private computePoolId(slot: bigint): string {
```

Comment:
Looks like this is no longer being used. Any reason to keep it?

Reply:
Yes I'd like to keep it, because it will be used to generate metadata